src/Features/Core/Portable/Diagnostics/HostAnalyzerManager.cs

```diff
@@ -154,6 +160,17 @@ private ImmutableArray<DiagnosticDescriptor> GetDiagnosticDescriptorsCore(Diagno
             return descriptors;
         }
 
+        /// <summary>
+        /// Returns cached type name for the given <paramref name="analyzer"/>.
+        /// </summary>
+        public string GetAnalyzerTypeName(DiagnosticAnalyzer analyzer)
+        {
+            return _analyzerTypeNameCache.GetValue(analyzer, ComputeAnalyzerTypeName);
```

Comment:
😕 Why are we caching this? The runtime already caches the result of `GetType().Name`.